### PR TITLE
implement preliminary MIDI support and switch the sound library to SDL2_mixer 

### PR DESCRIPTION
**Pull request by jopadan**
midi files playing still don't reflect gamestate condition I've included some basic data structures for it.

### CODE REVIEW

**Comment by hanatos** [comment]
whoa! super cool, someone is interested in c3 :)

thanks for looking into this! it's been a while since i had my fingers in this code, i shall take a look at your pr as soon as i get a minute.

File: include/triggers.h [hunk 1]
```diff
@@ -18,6 +18,29 @@
...
+  C3_COND_LOSE,
+  C3_COND_COMBAT,
+  C3_COND_PAD,
+  C3_COND_SCARY,
+}
+c3_condition_midi_t;
+
+static const char *c3_condition_midi_text[] =
+{
+  "dire.mid\0",
```
Comment (hanatos) [inline include/triggers.h]: i believe c adds the null byte for you if you don't explicitly allocate the array of chars short of one byte for the last character. the addition of the explicit "\0" is not needed.

**Comment by jopadan** [comment]
I'm still curious about the naming scheme of the midi files.
Is 'n' for night and how is it related to .mis music 0/1? is f for fm and g for general midi?

**Comment by hanatos** [comment]
very cool, thanks for this! i pushed it already, we can always polish later. noticed how terrible my flight dynamics are, will need to work on this. but it's already so much more awesome with music :)

**Comment by hanatos** [comment]
> I'm still curious about the naming scheme of the midi files.
> Is 'n' for night and how is it related to .mis music 0/1? is f for fm and g for general midi?

good question. my guess is that in the mis the "music" char is [abcde], the f and g prefix might be fm or general midi (related to the settings in setup.cfg?) and the "pad" "flight" "dire" .. parts are the game state. indeed i cannot find a line "N ;Music" in any of the mis, it might be night time. let me check night time missions :)

**Comment by hanatos** [comment]
quick check with c1m4.mis seems to indicate it plays gbpad on startup, not gnpad (even though it is a night time mission). so not at all sure what the n is about.

**Pull request merged**